### ISSUE TITLE
Applications are weirdly coloured

### ISSUE THREAD

**Issue by lboklin** [SimulaVR/Simula#54]
With or without HDR enabled, this happens:
```
QT_QPA_PLATFORM=wayland-egl konsole
QT_QPA_PLATFORM=wayland-egl kate
```
![image](https://user-images.githubusercontent.com/6375669/46580534-ea1ae800-ca26-11e8-87d2-9c1dbffc7e49.png)

**Comment by lboklin** [comment]
This seems to apply for all applications, regardless of whether run in XWayland or native Wayland.

**Comment by georgewsinger** [comment]
Red is blue, and blue is red:

![](https://i.imgur.com/mOJtQjr.png)

**Comment by lboklin** [comment]
So I experimented with swapping red and blue, and the colours are more correct but they appear washed out:
Simula:
![image](https://user-images.githubusercontent.com/6375669/48943149-8e4ae680-ef22-11e8-856c-cb92b85f4fd7.png)
vs desktop:
![image](https://user-images.githubusercontent.com/6375669/48943160-97d44e80-ef22-11e8-9557-f1ec45506e44.png)

So it doesn't appear to just be an BGRA8 -> RGBA8 issue.

**Comment by lboklin** [comment]
Should be noted that the only thing I'm doing is manipulating the colour of the pixels in the image *after* it has been created from the byte array.

**Comment by ChristophHaag** [comment]
Maybe qt is using a different color space, like sRGB vs RGBA? That would be one cause for "washed out" colors.

**Comment by lboklin** [comment]
It happens with any toolkit though.
Chromium in Simula:
![image](https://user-images.githubusercontent.com/6375669/48944226-239baa00-ef26-11e8-9e77-cb13e3f254e3.png)
and on desktop:
![image](https://user-images.githubusercontent.com/6375669/48944248-3ada9780-ef26-11e8-9d91-fea17d523f80.png)

**Comment by lboklin** [comment]
Regarding sRGB and RGBA, this is what the [Godot docs](https://godot.readthedocs.io/en/latest/classes/class_image.html) have to say about the format conversion:
> FORMAT_RGBA8 = 5 — OpenGL texture format RGBA with four components, each with a bitdepth of 8. Note that when creating an ImageTexture, an sRGB to linear color space conversion is performed.

**Comment by lboklin** [comment]
Fixed in d103d920fc72b87a298f28df2818fdb85d1fb222
